apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-plugin/src/test/java/co/elastic/apm/agent/opentelemetry/sdk/ElasticOpenTelemetryTest.java

```diff
@@ -0,0 +1,599 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.opentelemetry.sdk;
+
+import co.elastic.apm.agent.AbstractInstrumentationTest;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.ElasticContext;
+import co.elastic.apm.agent.impl.transaction.OTelSpanKind;
+import co.elastic.apm.agent.impl.transaction.Outcome;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ElasticOpenTelemetryTest extends AbstractInstrumentationTest {
+
+    private OpenTelemetry openTelemetry;
+    private Tracer otelTracer;
+
+    @Before
+    public void setUp() {
+        this.openTelemetry = GlobalOpenTelemetry.get();
+        assertThat(openTelemetry).isSameAs(GlobalOpenTelemetry.get());
+        otelTracer = openTelemetry.getTracer(null);
+
+        // otel spans are not recycled for now
+        disableRecyclingValidation();
```

Comment:
OK, so this answers my question above - we leak those on purpose.
Makes sense in general.
It looks like it will be quite difficult to maintain refs properly. Paying this extra overhead for arbitrary manual usage of the API worth it, but ideally we could also have a more optimized mode for libraries that embed OTel API usage.

Reply:
I think this answers my previous question on the distinction between manual instrumentation and calls to the OTel API that are already in an existing library.

Reply:
> I think this answers my previous question on the distinction between manual instrumentation and calls to the OTel API that are already in an existing library.

Just like happened to you, I only got here now 😄